presto-native-execution/presto_cpp/main/PrestoServer.cpp

```diff
@@ -155,14 +157,47 @@ PrestoServer::~PrestoServer() {}
 void PrestoServer::run() {
   auto systemConfig = SystemConfig::instance();
   auto nodeConfig = NodeConfig::instance();
-  int servicePort{0};
+  int httpPort{0};
   int httpExecThreads{0};
+
+  std::string certPath, keyPath, ciphers;
+  std::optional<int> httpsPort;
+
   try {
     systemConfig->initialize(
         fmt::format("{}/config.properties", configDirectoryPath_));
     nodeConfig->initialize(
         fmt::format("{}/node.properties", configDirectoryPath_));
-    servicePort = systemConfig->httpServerHttpPort();
+
+    httpPort = systemConfig->httpServerHttpPort();
+    if (systemConfig->enableHttps()) {
+      httpsPort = systemConfig->httpServerHttpsPort();
+
+      ciphers = systemConfig->httpsSupportedCiphers();
+      if (ciphers.empty()) {
+        VELOX_USER_FAIL("Https is enabled without ciphers")
+      }
+
+      auto otpionalCertPath = systemConfig->httpsCertPath();
+      if (!otpionalCertPath.has_value()) {
+        VELOX_USER_FAIL("Https is enabled without certificate path");
+      }
+      certPath = otpionalCertPath.value();
```

Comment:
Typo otpional -> optional

Reply:
Thanks. Let me submit another PR to fix.